code/modules/events/shuttle_catastrophe.dm

```diff
@@ -5,7 +5,7 @@
 	max_occurrences = 1
 
 /datum/round_event_control/shuttle_catastrophe/canSpawnEvent(players, gamemode)
-	if(SSshuttle.emergency.name == "Build your own shuttle kit")
+	if(istype(SSshuttle.emergency, /obj/docking_port/mobile/emergency/shuttle_build))
 		return FALSE //don't undo manual player engineering, it also would unload people and ghost them, there's just a lot of problems
 	if(EMERGENCY_AT_LEAST_DOCKED)
```

Comment:
Question why this doesn't trigger for BYOS

Reply:
Because BYOS' port is named "Shuttle Under Construction" and not "Build your own shuttle kit" and that's what SSshuttle.emergency is a pointer for. However while looking at this I realized there's an entirely different subtype for the build your own shuttle kit docking port and it makes much more sense to look at that subtype than look for a specific docking port name.

Reply:
I say about `if(EMERGENCY_AT_LEAST_DOCKED)`

Reply:
Oh. BYOS doesn't change that because that controls more than just the shuttle location, changing that would affect shuttle calling and recalling as well. If you wanted I suppose you could check the mode on the docking port, that would probably be closer to what you're thinking (so if the shuttle is somehow docked even when EMERGENCY_AT_LEAST_DOCKED is not true, it won't fire) but I can't think of anyway outside of serious adminbus for that to ever happen except for BYOS.

Reply:
Understand. I think about it like position of shuttle where `EMERGENCY_AT_LEAST_DOCKED` mean that it docked to station.
But it true for basic shuttle evacuation. BYOS placed to station in another way.

Reply:
Yes exactly.